external/synth/synth/core.py

```diff
@@ -163,6 +167,101 @@ def read_schema_from_zarr(
     return DatasetSchema(coord_schemes, variables)
 
 
+def read_directory_schema(rundir: str) -> Mapping[str, DatasetSchema]:
+    """Read schema from a directories
+
+    Does not support recursive directories or files other than zarr
```

Comment:
It would be nice if one could fully round trip all possible schemas written out by `write_dataset_schema` (including ones with non-zarr subdirectories and netCDF files), but that can probably be implemented later if we find we need it.

Reply:
I agree. I left out the netcdf parsing capability. Let's add that later.

Reply:
I added roundtripped read/write test.

Reply:
Sounds good; thanks.